Framework/Interaction/Rest/Company.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * ClassyLlama_AvaTax
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @copyright  Copyright (c) 2018 Avalara, Inc.
+ * @license    http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+
+namespace ClassyLlama\AvaTax\Framework\Interaction\Rest;
+
+use ClassyLlama\AvaTax\Api\RestCompanyInterface;
+use ClassyLlama\AvaTax\Framework\Interaction\Rest;
+use Magento\Framework\DataObject;
+
+class Company extends Rest implements RestCompanyInterface
+{
+    /**
+     * @param \Avalara\AvaTaxClient $client
+     * @param DataObject|null       $request
+     *
+     * @return mixed
+     * @throws \ClassyLlama\AvaTax\Exception\AvataxConnectionException
+     */
+    protected function getCompaniesFromClient( $client, $request = null )
+    {
+        if ($request === null)
+        {
+            $request = $this->dataObjectFactory->create();
+        }
+
+        $clientResult = $client->queryCompanies(
+            $request->getInclude(),
+            $request->getFilter(),
+            $request->getTop(),
+            $request->getSkip(),
+            $request->getOrderBy()
+        );
+
+        $this->validateResult( $clientResult, $request );
+
+        return $this->formatResult( $clientResult )->getValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getCompanies(
```

Comment:
@LordZardeck - Why do we have a getCompanies method _and_ a getCompaniesWithSecurity method?  Why would we ever attempt to fetch companies from the AvaTax API without authentication?  (That's not a valid way to use the endpoint anyway.)

Reply:
The whole point of `getCompaniesWithSecurity` was that at the config level, I'm needing to provide credentials that are not yet saved to the config. I wanted there to be a method that would retrieve them the way the rest of the REST methods are used, in addition, my one-off method that would override the credentials with those being entered on the config page.